xbmc/favourites/FavouritesService.cpp

```diff
@@ -95,9 +96,27 @@ bool IsMediasourceOfFavItemUnlocked(const std::shared_ptr<CFileItem>& item)
 
   return true;
 }
-} // anonymous namespace
 
-static bool LoadFromFile(const std::string& strPath, CFileItemList& items)
+std::string GetActionLabelFromPath(const std::string& path)
+{
+  const CURL url(path);
```

Comment:
Also you miss the run addon (probably others too):
![image](https://user-images.githubusercontent.com/7375276/195033167-11c90a4b-55e3-406f-8f66-c132d50407f1.png)

Makes me wonder if we really want to have this set on the item. Doesn't seem to bring any added value, we could simply hide the entry for favourites on the skin.

Reply:
> Also you miss the run addon

What do you mean? How does the exec string look like for this item? Must be "ActivateWindow(...:", otherwise I would display "Other / unknown". So, what's the problem here?

Reply:
> it exposes internal implementation details (the action of the item) which is of no use for the regular kodi user.

Well, I tend to disagree here. It is neither an implementation detail nor of no use for the user. If you think of a future GUI where you can manually add/edit favourites, IMO one of the most important fields to fill by the user would be the "action".

> Makes the code uglier

I don't see any "uglyness". Highly subjective opinion, though. ;-)

> is able to run headless you'd have to keep similar code either on the webinterface or on the Json-RPC API to mimic such behaviour.

I don't get this. Of course any GUI needs to implement everything that might be of interest for the user, using the available GUI labels. GUI designers can decide to use existing labels or not, but this particular label introduced here is absolutely nothing special, not forcing any future GUI to implement something.

Reply:
Personally anything that requires a long if/else series of conditions is ugly unless it's some factory method. Also binds translations/localizations to actions.
I see your point, maybe at some point we can mimic the same behaviour of super favourites in kodi with nodes and stuff like that (quite an usual feature request).

Can you at least move this to some sort of constexpr map (like that added by Lukas) or something similar that can avoid the long series of conditions?

Reply:
Not really, as I made it even worse and now display for "ActivateWindow" the name of the window. :-O

Let me put it this way: I will maintain that code and feel I can handle that bunch of conditions. I agree it is not the smartest approach, but actually is not really an issue, imo.

Reply:
lol, you won.